File: src/main/java/org/mifos/fineract/services/SelfLoanProductsApi.java
package org.mifos.fineract.services;

import retrofit2.http.GET;
import retrofit2.http.Headers;
import rx.Observable;

public interface SelfLoanProductsApi {
    /**
     * @param clientId (optional)
     * @return Observable&lt;String&gt;
     */
    @Headers({
            "Content-Type:application/json"
    })
    @GET("self/loanproducts")
    Observable<String> retrieveAllLoanProducts(
            @retrofit2.http.Query("clientId") Long clientId
    );

    /**
     * @param productId (required)
     * @param clientId  (optional)
     * @return Observable&lt;String&gt;
     */
    @Headers({
            "Content-Type:application/json"
    })
    @GET("self/loanproducts/{productId}")
    Observable<String> retrieveLoanProductDetails(
            @retrofit2.http.Path("productId") Long productId, @retrofit2.http.Query("clientId") Long clientId
    );

}

File: src/main/kotlin/org/openapitools/client/apis/SelfLoanProductsApi.kt
package org.openapitools.client.apis

import retrofit2.http.GET
import retrofit2.http.Path
import retrofit2.http.Query


interface SelfLoanProductsApi {
    /**
     *
     *
     * Responses:
     *  - 0: default response
     *
     * @param clientId  (optional)
     * @return [kotlin.String]
     */
    @GET("v1/self/loanproducts")
    suspend fun retrieveAllLoanProducts1(@Query("clientId") clientId: Long? = null): String

    /**
     *
     *
     * Responses:
     *  - 0: default response
     *
     * @param productId
     * @param clientId  (optional)
     * @return [kotlin.String]
     */
    @GET("v1/self/loanproducts/{productId}")
    suspend fun retrieveLoanProductDetails2(
        @Path("productId") productId: Long,
        @Query("clientId") clientId: Long? = null
    ): String

}
